backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/ddorder/process/WEBUI_MoveHUs_HUEditor_Launcher.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * #%L
+ * metasfresh-webui-api
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.ui.web.ddorder.process;
+
+import de.metas.handlingunits.IHUQueryBuilder;
+import de.metas.handlingunits.IHandlingUnitsDAO;
+import de.metas.handlingunits.model.I_DD_OrderLine;
+import de.metas.handlingunits.model.X_M_HU;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.ProcessExecutionResult;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.product.ProductId;
+import de.metas.ui.web.handlingunits.HUIdsFilterHelper;
+import de.metas.ui.web.process.adprocess.ViewBasedProcessTemplate;
+import de.metas.ui.web.view.CreateViewRequest;
+import de.metas.ui.web.view.IView;
+import de.metas.ui.web.view.IViewsRepository;
+import de.metas.ui.web.view.json.JSONViewDataType;
+import de.metas.util.Services;
+import org.compiere.SpringContextHolder;
+
+import static de.metas.ui.web.ddorder.HUsToMoveViewFactory.WINDOW_ID;
+import static de.metas.ui.web.ddorder.HUsToMoveViewFactory.WINDOW_ID_STRING;
+
+public class WEBUI_MoveHUs_HUEditor_Launcher extends ViewBasedProcessTemplate implements IProcessPrecondition
+{
+
+	public ProcessPreconditionsResolution checkPreconditionsApplicable()
+	{
+		if (getSelectedRowIds().isEmpty())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+		else if (getSelectedRowIds().isMoreThanOneDocumentId())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	private final IViewsRepository viewsRepo = SpringContextHolder.instance.getBean(IViewsRepository.class);
+
+	@Override protected String doIt() throws Exception
+	{
+		final I_DD_OrderLine ddOrderLine = getRecord(I_DD_OrderLine.class);
+
+		final IView husToMove = openHUEditor(ddOrderLine);
+
+		getResult().setWebuiViewToOpen(ProcessExecutionResult.WebuiViewToOpen.builder()
+				.viewId(husToMove.getViewId().getViewId())
+				.profileId(WINDOW_ID_STRING)
+				.target(ProcessExecutionResult.ViewOpenTarget.ModalOverlay)
+				.build());
+
+		return MSG_OK;
+	}
+
+	private IView openHUEditor(final I_DD_OrderLine orderLine)
```

Comment:
`createHUEditorView` would be  better than `openHUEditor` because it's not opening the view.

Reply:
agree